system/lib/standalone/standalone.c

```diff
@@ -155,24 +480,32 @@ double emscripten_get_now(void) {
   return (1000 * clock()) / (double)CLOCKS_PER_SEC;
 }
 
+__attribute__((__weak__))
+void _emscripten_throw_longjmp() {
+  _emscripten_err("The program tried to do an invalid longjmp, the program will exit now...");
+  abort();
+}
+
 // C++ ABI
 
 // Emscripten disables exception catching by default, but not throwing. That
 // allows users to see a clear error if a throw happens, and 99% of the
 // overhead is in the catching, so this is a reasonable tradeoff.
-// For now, in a standalone build just terminate. TODO nice error message
+// For now, in a standalone build just terminate.
 //
 // Define these symbols as weak so that when we build with exceptions
 // enabled (using wasm-eh) we get the real versions of these functions
 // as defined in libc++abi.
 
 __attribute__((__weak__))
 void __cxa_throw(void* ptr, void* type, void* destructor) {
+  _emscripten_err("The program tried to throw an exception, this is not supported in standalone mode, the program will exit now...");
   abort();
 }
 
 __attribute__((__weak__))
 void* __cxa_allocate_exception(size_t thrown_size) {
+  _emscripten_err("The program tried to allocate an exception, this is not supported in standalone mode, the program will exit now...");
```

Comment:
Can you use macro like in `emscripten_syscall_stubs.c`: https://github.com/emscripten-core/emscripten/blob/a197a6bd0dd31cdeea60dd421dc86c1cf9a0bf80/system/lib/libc/emscripten_syscall_stubs.c#L40-L44


Adding and using that macro could be its own small change?

Reply:
These kind of messages are normally not included (by emscripten) in release builds.    I don't see why these cases should be any different.

As well as enforcing consistency the macro name also gives a some semantic issue to the reader.  Its makes it easy to search for too.

Reply:
Alright. In my opinion you would want to make it clear why the program aborted if it's because of something that's not supported in standalone. Only doing abort doesn't really give any other info.

But I'll change it to a macro.

Reply:
Agreed that the abort on its own is now very useful, but there are lots of places in emscripten where the release build is unhelpful or less helpful, but the debug build is more helpful.    This keeps the release builds small, and (I hope) that developers are use to building a debug build to diagnose issues.

Reply:
I added the macro.